File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;
import entity.Result;

public interface BrandService {
	
	/**
	 * 查询所有的品牌
	 * @return
	 */
	List<TbBrand> findAll();
	
	/**
	 * 分页查询
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	PageResult findPage(int pageNum,int pageSize);
	
	/**
	 * 添加商品
	 * @param brand
	 * @return
	 */
	public void addBrand(TbBrand brand);
	
	/**
	 * 根据ID查询单个品牌
	 * @param id
	 */
	public TbBrand findOne(long id);
	
	/**
	 * 修改品牌信息
	 * @param brand
	 */
	public void updateBrand(TbBrand brand);
	
	/**
	 * 删除一个或多个品牌
	 * @param ids
	 * @return
	 */
	public void deleteBrands(long[] ids);
	
	/**
	 * 分页条件查询
	 * @return
	 */
	public PageResult findPage(TbBrand brand,int pageNum,int pageSize );
	
	/**
	 * 模块管理的查询下拉列表
	 */
	public List<Map> findOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){
		
		//查询所有品牌
		this.findAll = function(){
			return $http.get('../brand/findAll.do');
		}
		
		//分页查询
		this.search = function(page,rows,searchEntity){
			return $http.post('../brand/search.do?page='+page+'&rows='+rows,searchEntity);
		}
		
		//添加品牌
		this.add = function(entity){
			return $http.post('../brand/addBrand.do',entity);
		}
		
		//更新品牌
		this.update = function(entity){
			return $http.post('../brand/updateBrand.do',entity);
		}
		
		//根据ID查询品牌
		this.findOne = function(id){
			return $http.get('../brand/findById.do?id='+id);
		}
		
		//删除品牌
		this.delete = function(ids){
			return $http.get('../brand/deleteBrand.do?ids='+ids);
		}
	});